Intersect (Core)/Threading/ArgumentTypeChecker.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace Intersect.Threading
+{
+    /// <summary>
+    /// Utility class for checking the type of arguments passed to a new thread.
+    /// </summary>
+    /// <typeparam name="TArgument">Type of the argument that passes to a new thread.</typeparam>
+    public static class ArgumentTypeChecker<TArgument>
+    {
+        /// <summary>
+        /// Checks the type of each element within `args` array against the expected type `TArgument`. 
+        /// If any element in the array is not of type `TArgument` -> An exception is thrown.
+        /// </summary>
+        /// <param name="args">Array of arguments to be checked.</param>
+        public static void CheckArgumentTypes(object[] args)
+        {
+            if (args == null || args.Length < 1)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(args), $@"Expected one argument of type {typeof(TArgument).FullName}."
+                );
+            }
+
+            if (!(args[0] is TArgument))
+            {
+                throw new ArgumentException(
+                    $"Expected argument of type {typeof(TArgument).FullName}, but got {args[0].GetType().FullName}."
+                );
+            }
+        }
+    }
```

Comment:
I have two suggestions, one is significantly more work.

One is via generics, but that would need more than 1 method, 1 per X number of arguments.

The other one method where the consumer has to pass in an array of types.
- If either array is null, throw ArgumentNullException
- If the number of arguments is not exactly equal to the number of types, throw ArgumentException stating length mismatch, saying what the expected length is and what the actual length is
- Loop through all of the arguments matching them against the respective type for that argument index, if there is a type mismatch (using `is`, not `GetType()` just to be clear), throw stating what index it occurred at, what the expected type was, and what the argument type actually is